### DISCUSSION TITLE
Install Khartis online on my personal server.

### DISCUSSION THREAD

**Discussion by emerson-pj** [AtelierCartographie/Khartis#32] [Q&A]
Good afternoon.
I would like to know if it is possible to install the online version of Khartis on my personal server.
Installation similar to DSpace or OJS or Joomla.
It would be possible ?
How to make ?

**Comment by boogheta** [comment]
Hello,
Khartis is a simple static web application so it can be deployed pretty much on any web server, you just need to build it using this repository and following the documentation and use some server software such as nginx or apache to serve its build directory

**Comment by emerson-pj** [comment]
Hi,
I think I'm making a mistake in the build.
Would you help me ?

I am using for initial testing a computer with Windows 10 x64 and XAMPP v3.2.4 (apache v2.4).

I installed Git (Git-2.31.1-64-bit.exe)

Using Git CMD, I performed the steps below:

I downloaded it using:
git clone https://github.com/AtelierCartographie/Khartis.git
and put it in C:\xampp\htdocs\Khartis

I installed Node.js (node-v14.16.1-x64.msi)

Using Git CMD, I performed the steps below:
I used the command:
npm install

He returned the messages:
up to date, audited 3191 packages in 10s
72 packages are looking for funding
  run `npm fund` for details
73 vulnerabilities (28 low, 14 moderate, 28 high, 3 critical)
To address issues that do not require attention, run:
  npm audit fix
To address all issues possible (including breaking changes), run:
  npm audit fix --force
Some issues need review, and may require choosing
a different dependency.
Run `npm audit` for details.

So I used the command:
npm audit fix --force

He returned the messages:
.
.
.
74 vulnerabilities (20 low, 18 moderate, 29 high, 7 critical)

I tried to run:
ember serve

He returned the messages:
DEPRECATION: ember-cli-babel 5.x has been deprecated. Please upgrade to at least ember-cli-babel 6.6. Version 5.2.8 located: khartis -> ember-cli-codemirror-shim -> ember-cli-babel
DEPRECATION: ember-cli-babel 5.x has been deprecated. Please upgrade to at least ember-cli-babel 6.6. Version 5.2.8 located: khartis -> ember-cli-markdown-compiler -> ember-cli-babel
DEPRECATION: ember-i18n has been deprecated in favor of ember-intl
DEPRECATION: Overriding init without calling this._super is deprecated. Please call `this._super.init && this._super.init.apply (this, arguments);` addon: `release`
    at c: \ xampp \ htdocs \ Khartis \ node_modules \ ember-cli \ lib \ models \ project.js: 490: 40
.
.
.

I tried to open it in the browser http://localhost:4200/, but it didn't work.

I tried to open directly http://localhost/Khartis/, but it showed only the list of folders.

I tried to open http://localhost/Khartis/app, but returned the text "{{content-for" head "}} {{content-for" head-footer "}} {{content-for" body "}} { {content-for "body-footer"}} "


Where am I going wrong ?

**Comment by boogheta** [comment]
You need to follow the other steps listed in the readme: after doing npm install, you still need to copy and edit your configuration file and build the project using `ember build --environment production`

**Reply by emerson-pj** [reply]
Hi,
I ran the command above (`ember build --environment production`).
He showed the message:
node_modules appears empty, you may need to run `npm install`.

I ran the build from within the project folder.
`cd c: \ xampp \ htdocs \ khartis`
`ember build --environment production`
node_modules appears empty, you may need to run `npm install`.

But I have already run `npm install `and the node_modules folder is full of folders and files.
Location: `C: \ xampp \ htdocs \ khartis \ node_modules`

What could it be ?
Does the project necessarily need a Linux server?
Or am I missing something?

**Comment by boogheta** [comment]
It seems like a Windows specificity on how npm and ember work under Windows, which I have abolutely no knowledge nor experience with unfortunately. I can only advise you to search on the web for examples of using npm and ember under Windows sorry

**Reply by emerson-pj** [reply]
Hey, The error must be in Windows. I searched a lot on the internet. Others are having problems with Windows. I'm going to set up a Linux server and install Khartis on it. When I finish on Linux I notice it here. I appreciate your help.
